amundsen_application/static/js/components/SearchPage/index.tsx

```diff
@@ -91,6 +92,18 @@ export class SearchPage extends React.Component<SearchPageProps, SearchPageState
     }
   }
 
+  componentDidUpdate(prevProps) {
+    if (this.props.location !== prevProps.location) {
```

Comment:
It looks like we have some shared logic here between `componentDidMount` and `componentDidUpdate`. Can we consolidate that into a separate method? (will require updating tests)

Reply:
One more comment, was this `if` statement intended to prevent this logic from being called twice on the initial render? (On initial render `componentDidMount` will get called, which makes changes to the component so `componetDidUpdate` will also fire). If that is the case, `location` is an object and `if (this.props.location !== prevProps.location)` could always evaluate to true. 

Can you confirm if this is the case? if so, we could do `if (this.props.location.search !== prevProps.location.search)` to compare the actual search strings

Reply:
Right, yeah the object comparison here isn't great - comparing search directly makes more sense